bpython/curtsiesfrontend/repl.py

```diff
@@ -1870,8 +1868,8 @@ def prompt_for_undo():
     def redo(self):
         if (self.redo_stack):
             temp = self.redo_stack.pop()
-            self.push(temp)
             self.history.append(temp)
+            self.push(temp)
```

Comment:
What was the significance of this, why is it important to push append to history first?

Reply:
Oh, right. It makes no difference the order of those two.

But, I meant to also append to all_logical_lines right there, which _does_ have to be before self.push(temp). Rearranging the order of self.history would keep those three steps in the same order as is in on_enter.

Adding that commit now to add all_logical_lines there.

Reply:
cool, seems fine to reorder for consistency